database/raft/raft.go

```diff
@@ -735,11 +736,16 @@ func (sv *Service) applyEntry(ent raftpb.Entry, writers map[string]chan bool) {
 			defer sv.stateMu.Unlock()
 			defer sv.stateCond.Broadcast()
 			sv.state.RemovePeerAddr(cc.NodeID)
+		default:
+			panic(fmt.Errorf("unknown confchange type: %v", cc.Type))
```

Comment:
This seems like a prudent thing to check for, but it's separate right? Sorry I just want to be extra careful with stable branch fixes.

Of course it should never hit this case, but this line might actually turn a silent failure into a crash, which might be the right thing to do but might actually be worse than silently ignoring the request? Either way I'd rather do it in a separate patch or not at all for this branch.